src/braket/pennylane_plugin/ahs_device.py

```diff
@@ -0,0 +1,519 @@
+        poll_timeout_seconds (float): Total time in seconds to wait for
+            results before timing out.
+        poll_interval_seconds (float): The polling interval for results in seconds.
+        shots (int): Number of executions to run to acquire measurements. Defaults to 100.
+        aws_session (Optional[AwsSession]): An AwsSession object created to manage
+            interactions with AWS services, to be supplied if extra control
+            is desired. Default: None
+
+    """
+
+    name = "Braket Device for AHS in PennyLane"
+    short_name = "braket.aws.ahs"
+
+    def __init__(
+            self,
+            wires: Union[int, Iterable],
+            device_arn: str,
+            s3_destination_folder: AwsSession.S3DestinationFolder = None,
+            *,
+            poll_timeout_seconds: float = AwsQuantumTask.DEFAULT_RESULTS_POLL_TIMEOUT,
+            poll_interval_seconds : float = AwsQuantumTask.DEFAULT_RESULTS_POLL_INTERVAL,
+            shots: int = AwsDevice.DEFAULT_SHOTS_QPU,
+            aws_session: Optional[AwsSession] = None,
+    ):
+        device = AwsDevice(device_arn, aws_session=aws_session)
+        user_agent = f"BraketPennylanePlugin/{__version__}"
+        device.aws_session.add_braket_user_agent(user_agent)
+        
+        super().__init__(wires=wires, device=device, shots=shots)
+
+        self._s3_folder = s3_destination_folder
+        self._poll_timeout_seconds = poll_timeout_seconds
+        self._poll_interval_seconds = poll_interval_seconds
+
+
+    @property
+    def hardware_capabilities(self):
+        """Dictionary of hardware capabilities for the hardware device"""
+        return dict(self._device.properties.paradigm)
+
+    @property
+    def settings(self):
+        """Dictionary of constants set by the hardware.
+
+        Used to enable initializing hardware-consistent Hamiltonians by saving
+        all the values that would need to be passed, i.e.:
+
+            >>> dev_remote = qml.device('braket.aws.ahs', wires=3)
+            >>> dev_pl = qml.device('default.qubit', wires=3)
+            >>> settings = dev_remote.settings
+            >>> H_int = qml.pulse.rydberg.rydberg_interaction(coordinates, **settings)
+
+        By passing the ``settings`` from the remote device to ``rydberg_interaction``, an ``H_int`` Hamiltonian
+        term is created using the constants specific to the hardware. This is relevant for simulating the hardware
+        in PennyLane on the ``default.qubit`` device.
+        """
+        return {"interaction_coeff": self._get_rydberg_c6()}
+
+    def _get_rydberg_c6(self):
+        """Get rydberg C6 and convert from rad/s m^6 (AWS units) to Mrad/s um^6 (PL simulation units)"""
+        c6 = float(self._device.properties.paradigm.rydberg.c6Coefficient)  # rad/s x m^6
+        c6 = 1e-6 * c6  # rad/s --> M rad/s
+        c6 = c6 * 1e36  # m^6 --> um^6
+        return c6
+
+    def create_ahs_program(self, evolution):
```

Comment:
Does this override a method from PennyLane core? If not, can it be made private? More generally, make anything in this file that _doesn't_ override a PennyLane method private.

Reply:
This is the only non-private method on the devices that isn't overriding something in PennyLane core. Having it this way allows users to call `aquila.create_ahs_program(pennylane_pulse_operator)` and inspect the resulting AHS program (which for the hardware device has been discretized). This addresses @peterkomar-aws 's point about allowing users access to the program to inspect and understand what is going on.

If there is a different way to do this that would be preferable I'm happy to make this method private and do something else, but I would like to preserve some kind of user-facing option to obtain the AHS program _before_ they run a full circuit and create a hardware task.